opm/autodiff/NonlinearSolver_impl.hpp

```diff
@@ -137,19 +138,28 @@ namespace Opm
 
         // ----------  Main nonlinear solver loop  ----------
         do {
-            // Do the nonlinear step. If we are in a converged state, the
-            // model will usually do an early return without an expensive
-            // solve, unless the minIter() count has not been reached yet.
-            iterReport = model_->nonlinearIteration(iteration, timer, *this, reservoir_state, well_state);
+            try {
+                // Do the nonlinear step. If we are in a converged state, the
+                // model will usually do an early return without an expensive
+                // solve, unless the minIter() count has not been reached yet.
+                iterReport = model_->nonlinearIteration(iteration, timer, *this, reservoir_state, well_state);
 
-            report += iterReport;
-            report.converged = iterReport.converged;
+                report += iterReport;
+                report.converged = iterReport.converged;
 
-            converged = report.converged;
-            iteration += 1;
+                converged = report.converged;
+                iteration += 1;
+            }
+            catch (...) {
+                failureReport_ += report;
```

Comment:
yep, that's the idea: it only takes one failed iteration to make the previous iterations of the time step to be counted as failed as well. comment added.